lms/djangoapps/completion/waffle.py

```diff
@@ -4,23 +4,93 @@
+from openedx.core.djangoapps.waffle_utils import CourseWaffleFlag, WaffleFlagNamespace, WaffleSwitchNamespace
 
 # Namespace
 WAFFLE_NAMESPACE = 'completion'
+WAFFLE_FLAG_NAMESPACE = WaffleFlagNamespace(name='completion')
 
 # Switches
-
 # Full name: completion.enable_completion_tracking
 # Indicates whether or not to track completion of individual blocks.  Keeping
 # this disabled will prevent creation of BlockCompletion objects in the
 # database, as well as preventing completion-related network access by certain
 # xblocks.
 ENABLE_COMPLETION_TRACKING = 'enable_completion_tracking'
 
+# Full name completion.enable_visual_progress
+# Overrides completion.enable_course_visual_progress
+# Acts as a global override -- enable visual progress indicators
+# sitewide.
+ENABLE_VISUAL_PROGRESS = 'enable_visual_progress'
+
+# Full name completion.enable_course_visual_progress
+# Acts as a course-by-course enabling of visual progress
+# indicators, e.g. updated 'resume button' functionality
+ENABLE_COURSE_VISUAL_PROGRESS = 'enable_course_visual_progress'
+
+# SiteConfiguration visual progress enablement
+ENABLE_SITE_VISUAL_PROGRESS = 'enable_site_visual_progress'
+
 
 def waffle():
     """
     Returns the namespaced, cached, audited Waffle class for completion.
     """
     return WaffleSwitchNamespace(name=WAFFLE_NAMESPACE, log_prefix='completion: ')
+
+
+def waffle_flag():
+    """
+    Returns the namespaced, cached, audited Waffle flags dictionary for Completion.
+    """
+    namespace = WaffleFlagNamespace(name=WAFFLE_NAMESPACE, log_prefix=u'completion: ')
+    return {
+        # By default, disable visual progress. Can be enabled on a course-by-course basis.
+        # And overridden site-globally by ENABLE_VISUAL_PROGRESS
+        ENABLE_COURSE_VISUAL_PROGRESS: CourseWaffleFlag(
+            namespace,
+            ENABLE_COURSE_VISUAL_PROGRESS,
+            flag_undefined_default=False
+        )
+    }
+
+
+def visual_progress_enabled(course_key):
+    """
+    Exposes varia of visual progress feature.
+        ENABLE_COMPLETION_TRACKING, current_site.configuration, AND
+        enable_course_visual_progress OR enable_visual_progress
+
+    :return:
+
+        bool -> True if site/course/global enabled for visual progress tracking
+    """
+    if not waffle().is_enabled(ENABLE_COMPLETION_TRACKING):
+        return
+
+    if not site_configuration_enabled():
+        return
+
+    # Site-aware global override
+    if not waffle().is_enabled(ENABLE_VISUAL_PROGRESS):
+        # Course enabled
+        return waffle_flag()[ENABLE_COURSE_VISUAL_PROGRESS].is_enabled(course_key)
+
+    return True
+
+
+def site_configuration_enabled():
+    """
+    Helper function to return site-aware feature switch
+    :return: bool -> True if site enabled for visual progress tracking
+    """
+    try:
+        current_site = get_current_site()
+        if not current_site.configuration.get_value(ENABLE_SITE_VISUAL_PROGRESS, True):
```

Comment:
The default was changed back to `True` - this means that `site_configuration_enabled()` will return `True` if the `ENABLE_SITE_VISUAL_PROGRESS` key isn't present in the set of site config values for the current site.  Is that the intention, here?

Reply:
@iloveagent57 Right, we went through this. Fixing.

Reply:
That'll teach me to try to work in multiple code bases at the same time. Fixed now.